### PR TITLE
caper: init at 0.9

### PR DESCRIPTION
**Pull request by spaghetus**
## Description of changes

Caper is a tool for understanding and processing "pcap expressions" (also known as *tcpdump filters*) which are used for network packet analysis.
Caper can be used for:
* Expanding out pcap expressions "in full" to understand their implicit features.
* Reasoning about whether two expressions accept the same set of packets, or how their accepted packets differ.
* Converting pcap expressions into BPF programs.
* Converting between pcap expressions and English.

More info can be found in the Caper paper (https://www.nik.network/caper/pcap_semantics.pdf).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ca/caper/package.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  ];
+
+  buildPhase = ''
+    CAPER_WITH_ENGLISH=yes bash build.sh caper.native
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp caper.native $out/bin/caper
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/caper/package.nix:39]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
    cp caper.native $out/bin/caper
    runHook postInstall
  '';
```
Same for buildPhase

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by FliegendeWurst** [comment]
Please squash all package related commits into one.

File: pkgs/by-name/ca/caper/package.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    owner = "niksu";
+    repo = "caper";
+    rev = "v${version}";
+    hash = "sha256-TSryjz0NrGdkc+6vmfBqsuVpV3N9FvteTFsVqpUcm0w=";
+  };
+
+  nativeBuildInputs = [ ocaml ];
+
+  buildInputs = [
+    ocamlPackages.ocamlbuild
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/caper/package.nix:23]:
```suggestion
  nativeBuildInputs = [
    ocaml
    ocamlPackages.ocamlbuild
  ];

  buildInputs = [
```
It is used during build, not runtime.

File: pkgs/by-name/ca/caper/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    ocaml
+    ocamlPackages.ocamlbuild
+  ];
+
+  buildInputs = [
+    ocamlPackages.findlib
+    ocamlPackages.menhir
+    ocamlPackages.angstrom
+    gnum4
+  ];
```
Comment (FliegendeWurst) [inline pkgs/by-name/ca/caper/package.nix:30]:
```suggestion
  nativeBuildInputs = [
    ocaml
    ocamlPackages.ocamlbuild
    ocamlPackages.findlib
    ocamlPackages.menhir
    gnum4
  ];

  buildInputs = [
    ocamlPackages.angstrom
  ];

  strictDeps = true;
```
After this change the PR can be merged in my opinion.
(the last line ensures build tools and actual dependencies are cleanly separated)

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**